crates/router/src/connector/ebanx/transformers.rs

```diff
@@ -1,237 +1,406 @@
+    PixKey,
+}
+
+#[cfg(feature = "payouts")]
+#[derive(Debug, Serialize, Clone)]
+pub struct EbanxPayoutDetails {
+    name: Secret<String>,
+    email: Option<Email>,
+    document: Option<Secret<String>>,
+    document_type: Option<EbanxDocumentType>,
+    bank_info: EbanxBankDetails,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct EbanxCard {
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+#[cfg(feature = "payouts")]
+#[derive(Debug, Serialize, Clone)]
+pub enum EbanxDocumentType {
+    #[serde(rename = "CPF")]
+    NaturalPersonsRegister,
+    #[serde(rename = "CNPJ")]
+    NationalRegistryOfLegalEntities,
 }
 
-impl TryFrom<&EbanxRouterData<&types::PaymentsAuthorizeRouterData>> for EbanxPaymentsRequest {
+#[cfg(feature = "payouts")]
+#[derive(Debug, Serialize, Clone)]
+pub struct EbanxBankDetails {
+    bank_name: Option<String>,
+    bank_branch: Option<String>,
+    bank_account: Option<Secret<String>>,
+    account_type: Option<EbanxBankAccountType>,
+}
+
+#[cfg(feature = "payouts")]
+#[derive(Debug, Serialize, Clone)]
+pub enum EbanxBankAccountType {
+    #[serde(rename = "C")]
+    CheckingAccount,
+}
+
+#[cfg(feature = "payouts")]
+impl TryFrom<&EbanxRouterData<&types::PayoutsRouterData<api::PoCreate>>>
+    for EbanxPayoutCreateRequest
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: &EbanxRouterData<&types::PaymentsAuthorizeRouterData>,
+        item: &EbanxRouterData<&types::PayoutsRouterData<api::PoCreate>>,
     ) -> Result<Self, Self::Error> {
-        match item.router_data.request.payment_method_data.clone() {
-            domain::payments::PaymentMethodData::Card(req_card) => {
-                let card = EbanxCard {
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
+        let ebanx_auth_type = EbanxAuthType::try_from(&item.router_data.connector_auth_type)?;
+        match item.router_data.get_payout_method_data()? {
+            PayoutMethodData::Bank(Bank::Pix(pix_data)) => {
+                let bank_info = EbanxBankDetails {
+                    bank_account: Some(pix_data.bank_account_number),
+                    bank_branch: pix_data.bank_branch,
+                    bank_name: pix_data.bank_name,
+                    account_type: Some(EbanxBankAccountType::CheckingAccount),
+                };
+
+                let billing_address = item.router_data.get_billing_address()?;
+                let customer_details = item.router_data.request.get_customer_details()?;
+
+                let document_type = pix_data.tax_id.clone().map(|tax_id| {
+                    if tax_id.clone().expose().len() == 11 {
```

Comment:
Is this the correct and robust way to deal?

Reply:
the tax id comes in a specific format of fixed length, thus, this should be able to handle every cases, you can refer to this article for more [info](https://wise.com/gb/blog/cpf-cnpj-meaning-brazil#:~:text=What%20is%20the%20difference%20between,at%20the%20time%20of%20formation.)